test/Generators/Microsoft.Gen.Logging/Generated/LogPropertiesTests.cs

```diff
@@ -313,6 +313,27 @@ public void LogPropertiesInTemplateTest()
         _logger.Collector.LatestRecord.StructuredState.Should().NotBeNull().And.Equal(expectedState);
     }
 
+    // [Fact(Skip = "https://github.com/dotnet/extensions/issues/5411")]
+    [Fact]
+    public void LogPropertiesTestNullablePropertyInClass()
+    {
+        var now = new DateTime(2024, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+        var classToLog = new MyClassWithNullableProperty { NullableDateTime = now, NonNullableDateTime = now };
+        LogMethodNullablePropertyInClassMatchesNonNullable(_logger, classToLog);
+        Assert.Equal(1, _logger.Collector.Count);
+        Assert.Equal(LogLevel.Information, _logger.Collector.LatestRecord.Level);
+        Assert.Equal($"Testing nullable property within class here...", _logger.Collector.LatestRecord.Message);
+
+        var expectedState = new Dictionary<string, FakeLogRecord.AnnotatedStateValue>
+        {
+            ["classWithNullablePropertyParam.NullableDateTime"] = new FakeLogRecord.AnnotatedStateValue(typeof(DateTime), now.ToString(CultureInfo.InvariantCulture)),
+            ["classWithNullablePropertyParam.NonNullableDateTime"] = new FakeLogRecord.AnnotatedStateValue(typeof(DateTime), now.ToString(CultureInfo.InvariantCulture)),
+            ["{OriginalFormat}"] = new FakeLogRecord.AnnotatedStateValue(typeof(string), "Testing nullable property within class here...")
+        };
+
+        _logger.Collector.LatestRecord.AnnotatedState.Should().NotBeNull().And.Equal(expectedState);
```

Comment:
Instead of this, can we use the VerifyXunit snapshot verification?
```suggestion
        await Verifier.Verify(_logger.Collector.LatestRecord.AnnotatedState);
```